extensions/integration-tests/setEnvironmentVariables.js

```diff
@@ -0,0 +1,141 @@
+const KeyVault = require('azure-keyvault');
+const child_process = require('child_process');
+
+// Name of the values that are stored in Azure Key Vault
+const AKV_URL = 'https://sqltoolssecretstore.vault.azure.net/';
+const SECRET_AZURE_SERVER = 'ads-integration-test-azure-server';
+const SECRET_AZURE_SERVER_USERNAME = 'ads-integration-test-azure-server-username';
+const SECRET_AZURE_SERVER_PASSWORD = 'ads-integration-test-azure-server-password';
+const SECRET_BDC_SERVER = 'ads-integration-test-bdc-server';
+const SECRET_BDC_SERVER_USERNAME = 'ads-integration-test-bdc-server-username';
+const SECRET_BDC_SERVER_PASSWORD = 'ads-integration-test-bdc-server-password';
+const SECRET_STANDALONE_SERVER = 'ads-integration-test-standalone-server';
+const SECRET_STANDALONE_SERVER_USERNAME = 'ads-integration-test-standalone-server-username';
+const SECRET_STANDALONE_SERVER_PASSWORD = 'ads-integration-test-standalone-server-password';
+
+// Environment variable names
+const ENVAR_AZURE_SERVER = 'AZURE_SQL';
+const ENVAR_AZURE_SERVER_USERNAME = 'AZURE_SQL_USERNAME';
+const ENVAR_AZURE_SERVER_PASSWORD = 'AZURE_SQL_PWD';
+const ENVAR_BDC_SERVER = 'BDC_BACKEND_HOSTNAME';
+const ENVAR_BDC_SERVER_USERNAME = 'BDC_BACKEND_USERNAME';
+const ENVAR_BDC_SERVER_PASSWORD = 'BDC_BACKEND_PWD';
+const ENVAR_STANDALONE_SERVER = 'STANDALONE_SQL';
+const ENVAR_STANDALONE_SERVER_USERNAME = 'STANDALONE_SQL_USERNAME';
+const ENVAR_STANDALONE_SERVER_PASSWORD = 'STANDALONE_SQL_PWD';
+const ENVAR_PYTHON_INSTALL_PATH = 'PYTHON_TEST_PATH';
+
+// Mapping between AKV secret and the environment variable names
+const SecretEnVarMapping = [];
+SecretEnVarMapping.push([SECRET_AZURE_SERVER, ENVAR_AZURE_SERVER]);
+SecretEnVarMapping.push([SECRET_AZURE_SERVER_PASSWORD, ENVAR_AZURE_SERVER_PASSWORD]);
+SecretEnVarMapping.push([SECRET_AZURE_SERVER_USERNAME, ENVAR_AZURE_SERVER_USERNAME]);
+SecretEnVarMapping.push([SECRET_BDC_SERVER, ENVAR_BDC_SERVER]);
+SecretEnVarMapping.push([SECRET_BDC_SERVER_PASSWORD, ENVAR_BDC_SERVER_PASSWORD]);
+SecretEnVarMapping.push([SECRET_BDC_SERVER_USERNAME, ENVAR_BDC_SERVER_USERNAME]);
+SecretEnVarMapping.push([SECRET_STANDALONE_SERVER, ENVAR_STANDALONE_SERVER]);
+SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_PASSWORD, ENVAR_STANDALONE_SERVER_PASSWORD]);
+SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_USERNAME, ENVAR_STANDALONE_SERVER_USERNAME]);
+
+// Set the values that are not stored in AKV here
+process.env[ENVAR_PYTHON_INSTALL_PATH] = NOTEBOOK_PYTHON_INSTALL_PATH;
+
+const promises = [];
+
+// Fetch the values from AKV
+msrestAzure.interactiveLogin().then((credentials) => {
+	const client = new KeyVault.KeyVaultClient(credentials);
+	SecretEnVarMapping.forEach(entry => {
+		const secretName = entry[0];
+		const environmentVariable = entry[1];
+		const promise = client.getSecret(AKV_URL, secretName, '').then((result) => {
+			console.log(`${secretName}: ${result.value}`);
+			process.env[environmentVariable] = result.value;
```

Comment:
And to confirm - passing the values through an environment var is ok?

Reply:
yes, confirmed with Ken, this is the recommended way.